pkg/osquery/extension.go

```diff
@@ -330,14 +330,15 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 	})
 
 	// If no cached node key, enroll for new node key
+	// note that we set invalid two ways. Via the return, _or_ via isNodeInvaliderr
 	keyString, invalid, err := e.serviceClient.RequestEnrollment(ctx, e.Opts.EnrollSecret, identifier, enrollDetails)
 	if isNodeInvalidErr(err) {
 		invalid = true
 	} else if err != nil {
 		return "", true, errors.Wrap(err, "transport error in enrollment")
 	}
 	if invalid {
-		return "", true, errors.Wrap(err, "enrollment invalid")
+		return "", true, errors.Errorf("enrollment invalid (orig_err: %v)", err)
```

Comment:
don't we lose the backtrace etc if return a new error here?

Reply:
Yeah. I'm thinking I'm going to have to do something really ugly

Reply:
Okay. How's this look

Reply:
much better 😃